src/v1/device_manager_client.ts

```diff
@@ -815,23 +815,87 @@ export class DeviceManagerClient {
       {} | undefined
     ]
   > | void {
-    request = request || {};
-    let options: CallOptions;
-    if (typeof optionsOrCallback === 'function' && callback === undefined) {
-      callback = optionsOrCallback;
-      options = {};
-    } else {
-      options = optionsOrCallback as CallOptions;
-    }
-    options = options || {};
-    options.otherArgs = options.otherArgs || {};
-    options.otherArgs.headers = options.otherArgs.headers || {};
-    options.otherArgs.headers['x-goog-request-params'] =
-      this._gaxModule.routingHeader.fromParams({
-        parent: request.parent || '',
+    return new Promise(async (resolve, reject) => {
+      const token_response = await this.getRegistryToken();
+      const token = JSON.parse(token_response);
+      const payload = JSON.stringify({
+        id: request?.device?.id,
+        credentials: request?.device?.credentials        
+      })
+      var options = {
+        host: 'iot-sandbox.clearblade.com',
+        path: `/api/v/1/code/` + token.systemKey + `/devicesCreate`,
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'ClearBlade-UserToken': token.serviceAccountToken,
+          'Content-Length': payload.length
+        }
+      };
+
+      const req = https.request({
+        ...options,
+      }, res => {
+        let data = '';
+        const chunks: any[] = [];
+        res.on('data', chunk => data += chunk)
+        res.on('end', () => {
+          if (!this.isJsonString(data)) {
+            reject(data);
+            return;
+          }
+          let array: [protos.google.cloud.iot.v1.IDevice, protos.google.cloud.iot.v1.ICreateDeviceRequest | undefined, {} | undefined];
+          const deviceResponse = JSON.parse(data);
+          const icreatedevicerequest: protos.google.cloud.iot.v1.ICreateDeviceRequest | undefined = {};
```

Comment:
Hopefully we don't need to implement icreatedevicerequest. I haven't dug into how google handles this since it seems it's handled by their "innerApiCalls" which uses google-gax